File: solutions/2025-dec/24/markdown_image_parser.js
function parseImage(markdown) {
    const bracketStart = markdown.indexOf("[");
    const bracketEnd = markdown.indexOf("]");
    const altText = markdown.substring(bracketStart + 1, bracketEnd);

    const urlStart = markdown.indexOf("(");
    const urlEnd = markdown.indexOf(")");
    const urlText = markdown.substring(urlStart + 1, urlEnd);

    return `<img src="${urlText}" alt="${altText}">`;
}

File: solutions/2025-dec/24/markdown_image_parser.py
def parse_image(markdown):
    bracketStart = markdown.find("[")
    bracketEnd = markdown.find("]")
    altText = markdown[bracketStart + 1 : bracketEnd]

    urlStart = markdown.find("(")
    urlEnd = markdown.find(")")
    urlText = markdown[urlStart + 1 : urlEnd]

    return f'<img src="{urlText}" alt="{altText}">'
